nautobot/core/api/serializers.py

```diff
@@ -249,6 +264,51 @@ def to_internal_value(self, data):
             raise ValidationError(f"Related object not found using the provided ID: {pk}")
 
 
+class PolymorphicProxySerializer(_PolymorphicProxySerializer):
+    """
+    Like the base class from drf-spectacular, this is a pseudo-serializer used to represent multiple possibilities.
+
+    Use with `@extend_schema_field` on the method associated with a SerializerMethodField that can return one of
+    several different possible nested serializers. For example:
+
+        @extend_schema_field(
+            PolymorphicProxySerializer(
+                component_name="some_field",   # must be the same as the serializer field being decorated
+                serializers=[
+                    NestedSomeModelSerializer,
+                    NestedAnotherModelSerializer,
+                ],
+                allow_null=True,  # optional!
+            )
+        )
+        def get_some_field(self, obj):
+            ...
+
+    This has two key enhancements compared to the base class:
+
+    1. You can pass a callable (such as a lambda function) as the `serializers` argument, allowing for deferred
+       loading/initialization of the serializer list.
+    2. Supports `allow_null` as an init parameter, similar to real serializers.
+    """
+
+    @property
+    def serializers(self):
+        """Return the serializers provided for this instance, evaluating them first if needed."""
+        if callable(self._serializers):
+            self._serializers = self._serializers()
+        return self._serializers
+
+    @serializers.setter
+    def serializers(self, value):
+        """Set the serializers for this instance."""
+        self._serializers = value
```

Comment:
I contributed the serializers-as-callable feature upstream as https://github.com/tfranzel/drf-spectacular/pull/953, so once there's a new release we can remove some of this code.

Reply:
😎 https://github.com/tfranzel/drf-spectacular/releases/tag/0.26.0

Reply:
I have removed these methods and updated to `0.26` with 1b10237.

Since `allow_null` is not upstream I figured it would be preferred to be consistent to still use our overloaded class than split hairs between where we used ours vs what could use the default from DRF-Spectacular (example `origin` on `CablePathSerializer`).

Reply:
Yup, good stuff. I think the `allow_null` here doesn't quite work across the board, or at least that's what I was seeing when I put a little time last week into trying to port it upstream as well. (IIRC it works for our case where we're using a polymorphic serializer for a specific field on an API endpoint, but it didn't seem to work as expected for the upstream test case that has an entire API endpoint that's polymorphic.) So for the moment keeping our local `allow_null` enhancement here makes sense.